frontend/src/components/modals/TPluginInstallModal.tsx

```diff
@@ -0,0 +1,101 @@
+import { Component } from 'react';
+import { Translation } from 'react-i18next';
+
+import { InstallType } from '../../plugin';
+
+export enum TranslatedPart {
+  TITLE,
+  DESC,
+  BUTTON_IDLE,
+  BUTTON_PROC,
+}
+class TPluginInstallModal extends Component<TranslationProps> {
```

Comment:
Is there a way to do this without class components? We're trying to avoid using them as much as possible.

Reply:
This is a previous iteration without any additional components, but it looked ugly to me. If it's fine for you, I can revert this to that: https://github.com/RodoMa92/decky-loader/blob/91b2a1b38545bffd5d684e06e5ec9a1a20ca468c/frontend/src/components/modals/PluginInstallModal.tsx

Reply:
The additional component is fine, but I'd prefer a functional component instead of a class component.

Reply:
Fixed as of latest commit, replaced with a functional component.